controllers/envoy/revisions.go

```diff
@@ -272,7 +249,19 @@ func getRevisionIndex(version string, revisions []envoyv1alpha1.ConfigRevisionRe
 	return nil
 }
 
-func moveRevisionToLast(list []envoyv1alpha1.ConfigRevisionRef, idx int) []envoyv1alpha1.ConfigRevisionRef {
-
-	return append(list[:idx], append(list[idx+1:], list[idx])...)
+func getRevisionListOptions(namespace string, nodeID, version, envoyAPI *string) []client.ListOption {
+	labelSelector := client.MatchingLabels{}
+	if nodeID != nil {
+		labelSelector[nodeIDTag] = *nodeID
+	}
+	if version != nil {
+		labelSelector[versionTag] = *version
+	}
+	if envoyAPI != nil {
+		labelSelector[envoyAPITag] = *envoyAPI
+	}
+	return []client.ListOption{
+		labelSelector,
+		client.InNamespace(namespace),
+	}
 }
```

Comment:
This is a small helper to generate the list filters for the EnvoyConfigRevisions.